src/random.cpp

```diff
@@ -166,13 +166,12 @@ static void RandAddSeedPerfmon()
     if (ret == ERROR_SUCCESS) {
         RAND_add(vData.data(), nSize, nSize / 100.0);
         memory_cleanse(vData.data(), nSize);
-        LogPrint(BCLog::RAND, "%s: %lu bytes\n", __func__, nSize);
     } else {
-        static bool warned = false; // Warn only once
-        if (!warned) {
-            LogPrintf("%s: Warning: RegQueryValueExA(HKEY_PERFORMANCE_DATA) failed with code %i\n", __func__, ret);
-            warned = true;
-        }
+        // Performance data is only used as a last resort to improve entropy.
```

Comment:
The way I read the phrase "as a last resort" is: *if* there was a way to detect that our RNG attempts thus far have failed, we try this thing as a last resort. In the last resort also fails, that's critical. Hence the contraction.

But because we normally can't detect if we actually needed the last resort, we don't know if the failure was critical, and most likely it wasn't.

Reply:
Reformulated the text a bit.